Sync/SyncStateMachine.swift

```diff
@@ -73,10 +73,20 @@ open class SyncStateMachine {
     var stateLabelsSeen = [SyncStateLabel: Bool]()
     var stateLabelSequence = [SyncStateLabel]()
 
+    let stateLabelsIncluded: Set<SyncStateLabel>
+
     let scratchpadPrefs: Prefs
 
-    public init(prefs: Prefs) {
+    public convenience init(prefs: Prefs, optimistic: Bool = false) {
+        let included = optimistic ?
+            SyncStateLabel.optimisticValues :
+            SyncStateLabel.allValues
+        self.init(prefs: prefs, includedStates: included)
+    }
+
+    public init(prefs: Prefs, includedStates labels: [SyncStateLabel]) {
         self.scratchpadPrefs = prefs.branch("scratchpad")
+        self.stateLabelsIncluded = Set(labels)
```

Comment:
Is there a good reason to not just use a `Set` throughout?

(It'd be nice if Swift had `EnumSet` to make this cheap…)